Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean

```diff
@@ -91,38 +88,103 @@ lemma transpose_isTotallyUnimodular_iff (A : Matrix m n R) :
     Aᵀ.IsTotallyUnimodular ↔ A.IsTotallyUnimodular := by
   constructor <;> apply IsTotallyUnimodular.transpose
 
-lemma mapEquiv_isTotallyUnimodular {X' Y' : Type*} (A : Matrix m n R) (eX : X' ≃ m) (eY : Y' ≃ n) :
-    IsTotallyUnimodular ((A · ∘ eY) ∘ eX) ↔ A.IsTotallyUnimodular := by
-  rw [isTotallyUnimodular_iff, isTotallyUnimodular_iff]
-  constructor <;> intro hA k f g
-  · simpa [submatrix] using hA k (eX.symm ∘ f) (eY.symm ∘ g)
-  · simpa [submatrix] using hA k (eX ∘ f) (eY ∘ g)
+lemma IsTotallyUnimodular.reindex {A : Matrix m n R} (em : m ≃ m') (en : n ≃ n')
+    (hA : A.IsTotallyUnimodular) :
+    IsTotallyUnimodular (A.reindex em en) :=
+  hA.submatrix _ _
+
+lemma reindex_isTotallyUnimodular (A : Matrix m n R) (em : m ≃ m') (en : n ≃ n') :
+    IsTotallyUnimodular (A.reindex em en) ↔ A.IsTotallyUnimodular :=
+  ⟨fun hA => by simpa [Equiv.symm_apply_eq] using hA.reindex em.symm en.symm,
+   fun hA => hA.reindex _ _⟩
+
+-- TODO: move
```

Comment:
todo

Reply:
Yes, I mentioned it in the discussion thread. Ultimately, this TODO is Bhavik's call.

Reply:
I think we should just drop this lemma and inline it

Reply:
Yes, I agree with Eric